core/controllers/profile.py

```diff
@@ -358,7 +358,7 @@ def delete(self):
 class PendingAccountDeletionPage(base.BaseHandler):
     """The account pending deletion page."""
 
-    @acl_decorators.can_manage_own_profile
+    @acl_decorators.open_access
```

Comment:
Why open access? (Shouldn't it at least require login?)

Reply:
The problem is that we log out the user immediately after he tries to login and is scheduled to be deleted. After the logout, he is redirected to this page.

Reply:
OK -- maybe explain this in more depth in the docstring in line 359, so that it's clearer that this page doesn't require a user to be logged in, but is accessible by anyone, including users who have not requested deletion). 

Actually, speaking of which, should we find a way to redirect users away from this page if they haven't requested deletion of their account? I feel like it would be weird to show this to users who haven't done so, and it can easily happen (e.g. someone puts the wrong link in an email which doesn't match the text of the link). We could do something like: if we are doing the redirecting, we redirect to this page with a token; if this page is loaded and the token is invalid or nonexistent, we redirect them to the library page straightaway.

(Btw, in general, if a reviewer raises a question that needs explanation, consider adding that as a comment straightaway if you're doing other changes at the same time, since that could reduce a review cycle. The existence of such a question already indicates that at least one fellow developer doesn't understand the design choices.)

Reply:
Hmmm, I'm not sure if I understand this part:

> I feel like it would be weird to show this to users who haven't done so, and it can easily happen (e.g. someone puts the wrong link in an email that doesn't match the text of the link).

The redirect is only done to users who are scheduled to be deleted. So the only way the user can get to this page by accident is writing the address of this page into the URL.

Reply:
Or if they click on a link someone sends them, say.

Reply:
Yeah, but I still think it is a quite rare case, even the approach with the token could cause a similar problem because someone can send the link containing the token.

Reply:
Hm ok. 

For the token I was thinking of something like the CSRF token (i.e. it expires) but that does start to get a bit complicated.

So, let's leave it as you have it for now. I'll talk to Ben about this when we next meet to get a second opinion. Thanks!

Reply:
Okay, I have added info into the docstring.